games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,87 @@
+SUMMARY="The reference frontend of libretro"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media \
+players. It enables you to run classic games on a wide range of computers and \
+consoles through its slick graphical interface. It has advanced features like \
+shaders, netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com/"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_FILENAME="retroarch-$portVersion.tar.gz"
+SOURCE_DIR="RetroArch-$portVersion"
+
+srcGitRev_2="505ad6205e5fd8a155daad02ab39953a0b15b0a9"
+SOURCE_URI_2="https://github.com/libretro/retroarch-assets/archive/$srcGitRev_2.tar.gz"
+CHECKSUM_SHA256_2="de37772b00b755b80aafe5d4df0713c0fe6b0e5a79017d94e43f279e798aa85d"
+SOURCE_FILENAME_2="retroarch-assets-$portVersion.tar.gz"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	retroarch$secondaryArchSuffix = $portVersion
+	app:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch_cg2glsl$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libavcodec$secondaryArchSuffix
+	lib:libavformat$secondaryArchSuffix
+	lib:libavdevice$secondaryArchSuffix
+	lib:libswresample$secondaryArchSuffix
+	lib:libavresample$secondaryArchSuffix
+	lib:libavutil$secondaryArchSuffix
+	lib:libswscale$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+        cmd:gcc$secondaryArchSuffix
+        cmd:git
+	cmd:make
+        cmd:pkg_config$secondaryArchSuffix
+	cmd:which
+	"
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg keep-old
+"
+
+PATCH()
+{
+	libretroDir="`finddir B_SYSTEM_DIRECTORY`/$relativeLibDir/libretro"
+	retroarchDataDir="`finddir B_SYSTEM_DATA_DIRECTORY`/retroarch"
+	sed -i \
+		-e "s|^# \(libretro_directory =\)|\1 \"$libretroDir/\"|;" \
+		-e "s|^# \(libretro_info_path =\)|\1 \"$libretroDir/\"|;" \
+		-e "s|^# \(content_directory =\)|\1 \"$retroarchDataDir/\"|;" \
+		-e "s|^# \(assets_directory =\)|\1 \"$retroarchDataDir/assets/\"|;" \
+		retroarch.cfg
+}
+
+BUILD()
+{
+	ln -sfn $sourceDir2/retroarch-assets-$srcGitRev_2 media/assets
+	runConfigure --omit-dirs "binDir dataDir includeDir infoDir libDir libExecDir localStateDir oldIncludeDir sbinDir sharedStateDir sysconfDir" \
```

Comment:
Hmm, after all, it might be better to avoid the `runConfigure`.
So, here are the 2 possibilities:
* With `runConfigure`:
```
	runConfigure --omit-dirs "binDir dataDir includeDir infoDir libDir \
		libExecDir localStateDir oldIncludeDir sbinDir sharedStateDir \
		sysconfDir" \
		./configure --bindir="$appsDir" \
		--global-config-dir="$settingsDir" \
		--with-assets_dir="$dataDir"
```
* Without `runConfigure`:
```
	./configure --prefix="$prefix" --global-config-dir="$settingsDir" \
		--bindir="$appsDir" --datarootdir="$dataRootDir" \
		--docdir="$docDir" --mandir="$manDir" \
		--with-assets_dir="$dataDir"
```
Yep, I think the second option is slightly better. What do you think?

Note: I haven't tested the recipe yet, so there could be errors I haven't noticed.

Reply:
For what it's worth, I also believe that, since RetroArch's ./configure is entirely custom and not using autotools at all, the second option is simpler hence slightly better.